1/contrib/jenkins/kube-slave-common.sh

```diff
@@ -68,12 +68,21 @@ function get_is_names() {
   $oc_cmd get is -n "${PROJECT_NAME}" -l role=${SLAVE_LABEL} -o template --template "{{range .items}}{{.metadata.name}} {{end}}"
 }
 
+# get_istag_names returns a list of imagestreamtags that match the SLAVE_LABEL
+# role
+function get_istag_names() {
+  [ -z "$oc_cmd" ] && return
+  $oc_cmd get istag \
+     -n "${PROJECT_NAME}" \
+     -o 'go-template' \
+     --template="{{ range \$istag := .items }}{{ if \$istag.tag }}{{ if \$istag.tag.annotations }}{{ if index \$istag.tag.annotations \"role\"}}{{ if eq (index \$istag.tag.annotations \"role\") \"${SLAVE_LABEL}\"}}{{\$istag.metadata.name}}{{end}}{{end}}{{end}}{{end}}{{end}}"
```

Comment:
This can probably be:

```sh
$oc_cmd get istag \
    --namespace "${PROJECT_NAME}" \
    -o jsonpath="{ .items[?@.tag.annotations.role=='${SLAVE_LABEL}'].metadata.name }"
```

In general JSONPath will make all of these about 100000% clearer IMHO